cmd/juju/commands/bootstrap.go

```diff
@@ -273,43 +273,80 @@ func (c *bootstrapCommand) Info() *cmd.Info {
 		},
 		Purpose: usageBootstrapSummary,
 	}
-	if details := c.configDetails(); len(details) > 0 {
-		if output, err := common.FormatConfigSchema(details); err == nil {
-			info.Doc = fmt.Sprintf("%s%s\n%s%s",
-				usageBootstrapDetailsPartOne,
-				usageBootstrapConfigTxt,
-				output,
-				usageBootstrapDetailsPartTwo)
-			return jujucmd.Info(info)
-		}
-	}
-	info.Doc = strings.TrimSpace(fmt.Sprintf("%s%s",
-		usageBootstrapDetailsPartOne,
-		usageBootstrapDetailsPartTwo))
+	configKeys := c.configDetails()
 
+	info.Doc = fmt.Sprintf("%s%s\n%s%s",
+		usageBootstrapDetailsPartOne,
+		usageBootstrapConfigTxt,
+		configKeys.Format(),
+		usageBootstrapDetailsPartTwo)
 	return jujucmd.Info(info)
 }
 
-func (c *bootstrapCommand) configDetails() map[string]interface{} {
-	result := map[string]interface{}{}
-	addAll := func(m map[string]interface{}) {
-		for k, v := range m {
-			result[k] = v
-		}
+// ConfigCategoryKeys represents the collection of keys supported by the
+// --config option during bootstrap grouped by the domain category the keys
+// apply to within Juju.
+type ConfigCategoryKeys struct {
+	// BootstrapKeys describes the set of keys supported by --config as
+	// bootstrap only config keys.
+	BootstrapKeys map[string]common.PrintConfigSchema
+
+	// ControllerKeys describes the set of keys supported by --config as
+	// configuration items that will be applied to the controller during
+	// bootstrap.
+	ControllerKeys map[string]common.PrintConfigSchema
+
+	// ModelKeys describes the set of keys supported by --config as
+	// configuration items that will be applied to the controllers model during
+	// bootstrap.
+	ModelKeys map[string]common.PrintConfigSchema
```

Comment:
The list of ModelKeys, seems to be the same as ControllerKeys.

Reply:
Whoops. All fixed....